calico/felix/test/test_endpoint.py

```diff
@@ -53,18 +53,21 @@ def test_remove_deleted(self):
         """
         ep = endpoint.Endpoint(str(uuid.uuid4()), 'aa:bb:cc:dd:ee:ff')
 
-        with mock.patch('calico.felix.devices.tap_exists', \
-                        side_effect=[True, False]) as mock_exists,\
-             mock.patch('calico.felix.devices.list_tap_ips', \
-                         return_value = set(["1.2.3.4", "1.2.3.5"])) as mock_list, \
-             mock.patch('calico.felix.devices.del_route', \
-                        side_effect=futils.FailedSystemCall("blah",
-                                                            ["dummy", "args"],
-                                                            1,
-                                                            "",
-                                                            "")) as mock_del_route, \
-             mock.patch('calico.felix.frules.del_rules') as mock_del_rules:
-            ep.remove(iptables_state)
+        with mock.patch('calico.felix.devices.tap_exists',
+                        side_effect=[True, False]) as mock_exists:
+            with mock.patch(
+                'calico.felix.devices.list_tap_ips',
+                return_value = set(["1.2.3.4", "1.2.3.5"])) as mock_list:
+                with mock.patch(
+                    'calico.felix.devices.del_route',
+                    side_effect=futils.FailedSystemCall("blah",
+                                                        ["dummy", "args"],
+                                                        1,
+                                                        "",
+                                                        "")) as mock_del_route:
+                    with mock.patch(
+                        'calico.felix.frules.del_rules') as mock_del_rules:
+                        ep.remove(iptables_state)
```

Comment:
Is this a python 2.6 oddity that prevents you just having multiple "with" statements (which works for 2.7 so long as you use enough backslashes on the end of lines)? If so, then presumably there are a host of other places where this problem occurs, since I use that logic all the time in tests.

From: Cory Benfield [mailto:notifications@github.com]
Sent: 19 February 2015 17:31
To: Metaswitch/calico
Subject: Re: [calico] Remove python 2.7 only features so we work on python 2.6 again. (#209)


In calico/felix/test/test_endpoint.py<https://github.com/Metaswitch/calico/pull/209#discussion_r25006804>:

> -            ep.remove(iptables_state)

> +        with mock.patch('calico.felix.devices.tap_exists',

> +                        side_effect=[True, False]) as mock_exists:

> +            with mock.patch(

> +                'calico.felix.devices.list_tap_ips',

> +                return_value = set(["1.2.3.4", "1.2.3.5"])) as mock_list:

> +                with mock.patch(

> +                    'calico.felix.devices.del_route',

> +                    side_effect=futils.FailedSystemCall("blah",

> +                                                        ["dummy", "args"],

> +                                                        1,

> +                                                        "",

> +                                                        "")) as mock_del_route:

> +                    with mock.patch(

> +                        'calico.felix.frules.del_rules') as mock_del_rules:

> +                        ep.remove(iptables_state)

Oh yeah! I knew that was somewhere, but I could only see contextlib.ExitStack in recent docs so I got tunnel vision.

—
Reply to this email directly or view it on GitHub<https://github.com/Metaswitch/calico/pull/209/files#r25006804>.